In [136]:
import numpy as np
import pandas as pd

In [137]:
opm_df = pd.read_csv('data/OPM_playlist_tracks.csv')
pinoy_df = pd.read_csv('data/pinoy_playlist_tracks.csv')
manila_df = pd.read_csv('data/manila_playlist_tracks.csv')
tagalog_df = pd.read_csv('data/tagalog_playlist_tracks.csv')
bisaya_df = pd.read_csv('data/bisaya_playlist_tracks.csv')
ilocano_df = pd.read_csv('data/ilocano_playlist_tracks.csv')

In [138]:
filipino_df = pd.concat([opm_df, ilocano_df, bisaya_df, tagalog_df, manila_df, pinoy_df])

In [139]:
filipino_df

track_id                    name  popularity  duration_ms  \
0     3VcVQmMivq2ISfKUmIuvMK                 Rainbow          63       347120   
1     3OpGUlDmRUXh0NkIYWoIlD               Your Song          66       168493   
2     761frfZtZ8I2g70UsjFfBK   I Need You More Today          63       233728   
3     1yDiru08Q6omDOGkZMPnei         Maybe The Night          69       217197   
4     54b8IPvheXDpro2VR2rWAS                   Ivana          61       246550   
...                      ...                     ...         ...          ...   
1948  2BgD4nRyx9EZ5o8YEnjRSV  Kung 'Di Rin Lang Ikaw          66       267759   
1949  00mBzIWv5gHOYxwuEJXjOG   Sa Ngalan Ng Pag-Ibig          64       285138   
1950  1isVBafE79C4qRFLdia35T          On Bended Knee          51       328360   
1951  56srmLWNnQcMI7LpzyCNgR              Brown Eyes          45       273866   
1952  5V2AX2WTOHl2FKtftvisV6        Kahit Ayaw Mo Na           1       243015   

                                              artist_id  \
0                                6ZgBJRjJsK1iOJGKGZxEUv   
1                                2XHTklRsNMOOQT56Zm3WS4   
2                                3XEHRbR9NMWfNzQ6TQMS9M   
3                                4DAcJXcjX0zlQAZAPAx4Zb   
4                                2EGJbNf0Hva2C2N6hvhMXD   
...                                                 ...   
1948  ['4qFxP3qN9GsnZDOkAE6x2m', '0rZRTXEmmPmx6gt92t...   
1949                             4qFxP3qN9GsnZDOkAE6x2m   
1950                             11Jdq2gOTBhmfEYmFrZlue   
1951                             1Y8cdNmUJH7yBTd9yOvr5i   
1952                             5HIZU0JzM0AgfItVOm4E08   

                                  artist_name  num_artists  \
0                                South Border            1   
1                            Parokya Ni Edgar            1   
2                                Caleb Santos            1   
3                                     Ben&Ben            1   
4                                   Soulstice            1   
...                                       ...          ...   
1948  ['December Avenue', 'Moira Dela Torre']            2   
1949                          December Avenue            1   
1950                            Kris Lawrence            1   
1951                          Destiny's Child            1   
1952                                This Band            1   

                 playlist_id           playlist_name  
0     37i9dQZF1DX4olOMiqFeqU           OPM Favorites  
1     37i9dQZF1DX4olOMiqFeqU           OPM Favorites  
2     37i9dQZF1DX4olOMiqFeqU           OPM Favorites  
3     37i9dQZF1DX4olOMiqFeqU           OPM Favorites  
4     37i9dQZF1DX4olOMiqFeqU           OPM Favorites  
...                      ...                     ...  
1948  7qdNa0n7Kf31vfFor2gmdz  PINOY HITS SONGS 🔥🎶🔊🇵🇭  
1949  7qdNa0n7Kf31vfFor2gmdz  PINOY HITS SONGS 🔥🎶🔊🇵🇭  
1950  7qdNa0n7Kf31vfFor2gmdz  PINOY HITS SONGS 🔥🎶🔊🇵🇭  
1951  7qdNa0n7Kf31vfFor2gmdz  PINOY HITS SONGS 🔥🎶🔊🇵🇭  
1952  7qdNa0n7Kf31vfFor2gmdz  PINOY HITS SONGS 🔥🎶🔊🇵🇭  

[11318 rows x 9 columns]

In [140]:
chart_opm_df = chart_df[chart_df['track_id'].isin(filipino_df['track_id'])]
chart_opm_df.shape

(104494, 6)

In [141]:
chart_opm_artist_df = chart_opm_df.drop_duplicates(subset='track_id')
chart_opm_artist_df.shape

(440, 6)

In [142]:
chart_opm_artist_df=chart_opm_artist_df.drop_duplicates(subset='artist')
chart_opm_artist_df['artist']

24                     SUD
39        Silent Sanctuary
45          Up Dharma Down
49        Moira Dela Torre
50              Bruno Mars
                ...       
278697          Little Mix
279000       Ariana Grande
283559            JM Bales
291586            347aidan
293287      Olivia Rodrigo
Name: artist, Length: 220, dtype: object

In [143]:
chart_opm_artist_df['artist'].unique()

array(['SUD', 'Silent Sanctuary', 'Up Dharma Down', 'Moira Dela Torre',
       'Bruno Mars', 'Daryl Ong', 'Yeng Constantino',
       'Michael Pangilinan', 'Juris', 'Moonstar88', 'Nina', 'Mark Carpio',
       'MYMP', 'South Border', 'James Reid', '6cyclemind',
       'Julie Anne San Jose', 'Piolo Pascual', 'The Goo Goo Dolls',
       'Sitti', 'Bailey May', 'Eraserheads', 'Jericho Rosales',
       'Angeline Quinto', 'John Mayer', 'Christina Perri', 'Toni Gonzaga',
       'Hale', 'Morissette', 'Abra', 'Hoobastank', 'Coldplay', 'The Fray',
       'Joshua Garcia', 'KZ Tandingan', 'The Company', 'Callalily',
       'Gloc 9', 'Inigo Pascual', 'Sponge Cola', 'Christian Bautista',
       'Jona', 'Kiana V', 'Edwin McCain', 'Rachel Platten', 'Jay R',
       'Kaye Cal', 'krissy & ericka', 'The Chainsmokers', 'Erik Santos',
       'Sarah Geronimo', 'DJ Khaled', 'Claudia', 'Thyro', 'Mayonnaise',
       'Daniel Padilla', 'Louis Tomlinson', 'Parokya Ni Edgar',
       'Caleb Santos', 'Rivermaya', 'Dani

In [144]:
chart_opm_artist_df.to_csv("chart_opm_artist_to_be_verified.csv", index=False, encoding='utf-8')

OPM : 95
OPM + Ilocano : 100
OPM + Ilocano + Bisaya : 105
OPM + Ilocano + Bisaya + Tagalog: 133 (Some foreign singers)
OPM + Ilocano + Bisaya + Tagalog + Manila: 177 (more foreign singers)
OPM + Ilocano + Bisaya + Tagalog + Manila + Pinoy:220 (more foreign singers) Filipino is just 147
